web/src/main/java/org/springframework/security/web/authentication/AuthenticationSuccessHandler.java

```diff
@@ -38,6 +39,23 @@
  */
 public interface AuthenticationSuccessHandler {
 
+	/**
+	 * Called when a user has been successfully authenticated.
+	 *
+	 * @param request the request which caused the successful authentication
+	 * @param response the response
+	 * @param chain the {@link FilterChain} which can be used to proceed other filters in the chain
+	 * @param authentication the <tt>Authentication</tt> object which was created during
+	 * the authentication process.
+	 * @since 5.2.0
+	 */
+	default void onAuthenticationSuccess(HttpServletRequest request,
```

Comment:
I really like how you added a default method here. This is very powerful for our APIs, but still passive.